api/src/main/java/org/apache/gravitino/file/FilesetDataOperationCtx.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.file;
+
+import org.apache.gravitino.annotation.Evolving;
+
+/**
+ * An interface representing a fileset data operation context. This interface defines some
+ * information need to report to the server.
+ *
+ * <p>{@link FilesetDataOperationCtx} defines the basic properties of a fileset data operation
+ * context object.
+ */
+@Evolving
+public interface FilesetDataOperationCtx {
+  /** @return The sub path which is operated by the data operation . */
+  String subPath();
+
+  /** @return The data operation type. */
+  FilesetDataOperation operation();
+
+  /** @return The client type of the data operation. */
+  ClientType clientType();
+}
```

Comment:
@xloya my general feeling is that using "interfaces" to define the content of fileset operation is too strict, it is not convenient to add more properties unless we change the API definitions, DTOs and REST server/client.

My thinking is that we can simply using `Map<String, String>` to store the context information we want, that will be simple to extend, another thing is that we can refer to Hadoop's CallerContext to see how it design.

My feeling is that using a API system to define the caller context information seems too overkill, what do you think?

Reply:
@jerryshao Yes, I agree with you that using `Map<String, String>` is more scalable.
I briefly looked at the implementation of `CallerContext` in Hadoop. According to the description in the PR(https://issues.apache.org/jira/browse/HDFS-9184), there will be the following interaction process:
1. When the client initiates an RPC request to the server, it fills the context into the Header of the RPC request.
2. After receiving the request, the server obtains the context from the Header of the RPC request and performs an audit record.
This means that in this way, we can put these audit contents into the Header of the `getFilesetContext` API request, and only keep the `subPath` information in the Request Body to exchange Fileset metadata and real paths.
Do you think we should keep the context in the Request Body in the form of `Map<String, String>`, or put it in the Header?
In addition, the purpose of adding this API is not only for auditing, but also to unify the logic of converting and obtaining real paths in various language clients, which can make the logic in PyGVFS/Hadoop GVFS simpler.

Reply:
I understand the purpose of adding a new interface, what I'm thinking about is not just the `FileContext` API, I was thinking how to design this in a user-friendly way. 

The current interface here is a little hard for user to understand its purpose and usage. I'm more concerned about this, can you please write a document about how you design this, either in the current way, or using `map`, or even using caller context. And we can discuss based on this. Overall, I think we should be more cautious when the changes involves API definitions.

Reply:
@jerryshao I have wrote a document about how to pass audit content when accessing the API. Please check it out when you have time, thanks: https://docs.google.com/document/d/1wnSRSPcppBrJnb5MAL8gWuNNvhSkeXVXCE0r_zcYd1M.
In general, from the perspective of open source and commercial product practices, it is more common to put additional audit content in the HTTP Header of the API request. In addition, as far as the API in this PR is concerned,  considering that it only needs to be open to the Hadoop / Python GVFS client, we may consider not expose the API in the Java/Python Client, and only Hadoop/Python GVFS itself maintains the logic of calling the API. WDYT?